packages/persist/lib/controllers/persist.controller.ts

```diff
@@ -270,16 +270,16 @@ class PersistController {
                 content: `There was an issue with the batch ${persistType}. ${error?.message}`,
                 timestamp: Date.now()
             });
-            const errMsg = `Failed to ${persistType} records ${activityLogId}`;
-            span.setTag('error', errMsg).finish();
-            return err(errMsg);
+            const res = err(`Failed to ${persistType} records ${activityLogId}`);
+            span.setTag('error', res.error).finish();
```

Comment:
`tracer.setTag('error', e)` e must be an error, not a string. see https://docs.datadoghq.com/tracing/trace_collection/custom_instrumentation/dd_libraries/nodejs/?tab=errors